### DISCUSSION TITLE
Debug print in AppFactory.h

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#25767] [Q&A General] [answered]
Hello:

This one should be very simple.  I am tracking down a segfault caused when building an AuxKernel.  It would help if I could print out some of the parameters.  But, if I use "Moose::out << ...", I get the following error message:

In file included from /Users/joe/projects/SAM/moose/modules/fluid_properties/src/base/FluidPropertiesApp.C:12:
/Users/joe/projects/SAM/moose/framework/build/header_symlinks/AppFactory.h:42:10: error: no type named 'out' in namespace 'Moose'
  Moose::out << "AppFactoryBuildInfoBase" << std::endl;

This is a first.  So, I tried "std::out << ...." and included <iostream>

The compile error is:

In file included from /Users/joe/projects/SAM/moose/modules/fluid_properties/src/base/FluidPropertiesApp.C:12:
/Users/joe/projects/SAM/moose/framework/build/header_symlinks/AppFactory.h:42:8: error: no type named 'cout' in namespace 'std'
  std::cout << "AppFactoryBuildInfoBase" << std::endl;
  ~~~~~^

How can I put a debug print statement inside this template:

template <typename T>
struct AppFactoryBuildInfo : public AppFactoryBuildInfoBase
{
  std::cout << "AppFactoryBuildInfoBase" << std::endl;
  virtual MooseAppPtr build(const InputParameters & params) override
  {
    return std::make_shared<T>(params);
  }
  virtual InputParameters buildParameters() override { return T::validParams(); }
};

Thanks,
  Joe Kelly

**Comment by GiudGiud** [comment, marked as answer]
hi Joe

can you do this

```
template <typename T>
struct AppFactoryBuildInfo : public AppFactoryBuildInfoBase
{
virtual MooseAppPtr build(const InputParameters & params) override
{
  std::cout << "AppFactoryBuildInfoBase" << std::endl;
  return std::make_shared<T>(params);
}
```

we cant put a print statement inside the definition for the template, it s a place for declaring code not executing it

**Reply by joe61vette** [reply]
Thanks Guillaume.  When I put it inside the {}, Moose::out works.  THANKS!
